grammarinator/runtime/generator.py

```diff
@@ -26,6 +30,27 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._gen._max_depth += 1
 
 
+class AlternationContext(object):
+
+    def __init__(self, gen, min_depths, conditions):
+        self._gen = gen
+        self._min_depths = min_depths
+        self._conditions = conditions
+        self._orig_depth = gen._max_depth
+
+    def __enter__(self):
+        available_options = [w if self._min_depths[i] <= self._gen._max_depth else 0 for i, w in enumerate(self._conditions)]
+        if sum(available_options) > 0:
+            return available_options
```

Comment:
A similar minor naming issue here. Sometimes the weights are called conditions, sometimes choices, sometimes "alts" (see above). Now, the weights are "available options". It would be nice to use less terms for the same concept. Perhaps `available_weights` (or `filtered_weights`) would be a clearer name here.